code/cryptography/src/autokey_cipher/autokey_cipher.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include <string.h>
+
+void flush_stdin()
+{
+    while (fgetc(stdin) != '\n');
+}
+
+std::string autokeycipher(std::string text, char key, int encrypt)
+{
+    int textsize;
+    textsize = sizeof(text) / sizeof(text[0]);
```

Comment:
This is incorrect. Use `string.length()` and don't store it in a variable. Also, you need to pass the string as a reference, or else it won't be modified.